src/Stratis.Bitcoin/P2P/Peer/NetworkPeer.cs

```diff
@@ -198,6 +198,27 @@ public bool IsConnected
             }
         }
 
+        /// <inheritdoc />
+        public bool HasRemoteIPAddress(IPAddress ip, int? port = null)
```

Comment:
I'm still not sure on the naming... This name implies that this peer has a remote IP address?

Reply:
We want to know if a network peer matches another...

Reply:
The naming is to indicate that we are matching against the peer's remote ip address.

Reply:
Then maybe call it `MatchRemoteIPAddress`?

Reply:
Done.